app/src/main/java/ee/ioc/phon/android/speechutils/AbstractAudioRecorder.java

```diff
@@ -74,11 +74,16 @@ protected AbstractAudioRecorder(int audioSource, int sampleRate) {
         this(audioSource, sampleRate, DEFAULT_BUFFER_LENGTH_IN_MILLIS, false);
     }
 
-    protected void createRecorder(int audioSource, int sampleRate, int bufferSize) {
-        mRecorder = new SpeechRecord(audioSource, sampleRate, AudioFormat.CHANNEL_IN_MONO, RESOLUTION, bufferSize, false, false, false);
+    protected SpeechRecord createRecorder(int audioSource, int sampleRate, int bufferSize) {
+        if (mRecorder != null)
+            release();
+
+        mRecorder = new SpeechRecord(audioSource, sampleRate, AudioFormat.CHANNEL_IN_MONO, RESOLUTION, bufferSize, true, false, false);
```

Comment:
Why did you switch on the noise suppressor? Could you set it back to "false" unless there is a good reason for this change.